usr/src/uts/armv8/sys/smp_impldefs.h

```diff
@@ -41,8 +41,27 @@ extern "C" {
  *	External Reference Functions
  */
 extern int (*slvltovect)(int);	/* ipl interrupt priority level */
-extern int setlvl(int);	/* set intr pri represented by vect */
-extern void setlvlx(int);	/* set intr pri to specified level */
+
+/*
+ * Enter the context of the interrupt vector specified as the parameter.
+ *
+ * set interrupt priority to that of the interrupt vector parameter, and
+ * return the old interrupt priority.
```

Comment:
Does this have the same semantics with the  -1 return on spurious interrupt ala x86?

Reply:
No, but I can't say for sure it won't ever?

Reply:
This condition should mostly be caught (in a different way) before we hit this code - see the call to `gic_is_spurious` in `do_interrupt` (armv8/os/intr.c).